### PR TITLE
MGMT-3382: Retrieve OCP versions from API

### PR DESCRIPTION
**Pull request by mareklibra**
Depends on:
- [x] https://github.com/openshift/assisted-service/pull/869
- [x] https://github.com/openshift-assisted/assisted-ui-lib/pull/398 Api update for `/openshift_versions` endpoint

### CODE REVIEW

**Comment by mareklibra** [comment]
The warning is rendered conditionally - just for non-production releases.

File: src/components/clusters/NewClusterPage.tsx [hunk 1]
```diff
@@ -81,13 +85,21 @@ const NewClusterForm: React.FC<NewClusterFormProps> = ({ pullSecret = '' }) => {
     }
   };
 
+  const getOpenshiftVersionHelperText = (value: string) =>
+    versions.find((version) => version.value === value)?.supportLevel !== 'production' ? (
+      <>
+        <ExclamationTriangleIcon color={warningColor.value} size="sm" />
+        &nbsp;Please note that this version is not production ready.
```
Comment (mareklibra) [inline src/components/clusters/NewClusterPage.tsx:92]: Cc @andybraren , @nirfarkas

File: src/components/OpenshiftVersionsContextProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+export const OpenshiftVersionsContextProvider: React.FC = ({ children }) => {
+  const [versions, setVersions] = React.useState<OpenshiftVersionOptionType[]>([
+    // TODO(mlibra): provide empty array when /openshift_version API endpoint is deployed to production
+    DEFAULT_OPENSHIFT_VERSION,
+  ]);
+  React.useEffect(() => {
+    const doAsync = async () => {
+      try {
+        const { data } = await getOpenshiftVersions();
+        const versions: OpenshiftVersionOptionType[] = Object.getOwnPropertyNames(data)
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/OpenshiftVersionsContextProvider.tsx:35]: done

File: src/components/OpenshiftVersionsContextProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  `https://bugzilla.redhat.com/enter_bug.cgi?product=OpenShift%20Container%20Platform&Component=OpenShift%20Container%20Platform&component=assisted-installer&version=${version}`;
+
+export type OpenshiftVersionsContextType = {
+  versions: OpenshiftVersionOptionType[];
+
+  // Assumption: AddHostsClusterCreateParams['openshiftVersion'] is a subset of OpenshiftVersionOptionType['value'] and contains the latest supported versions
+  // In case this assumption is wrong, let's pass the "wrong" value to the Day 2 API and let it fail on server validation.
+  getDefaultVersion: () => AddHostsClusterCreateParams['openshiftVersion'];
+  normalizeClusterVersion: (version?: string) => AddHostsClusterCreateParams['openshiftVersion'];
+  getBugzillaLink: (version?: OpenshiftVersionOptionType['value']) => string;
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/OpenshiftVersionsContextProvider.tsx:16]: Recently the list of versions in BZ of the relevant component are the OCP versions. But you are right, we do not need to include that function in the context.

File: src/components/OpenshiftVersionsContextProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        setVersions(versions);
+      } catch (e) {
+        return handleApiError(e);
+        // keep using the default hardcoded versions
+      }
+    };
+    doAsync();
+  }, [setVersions]);
+
+  const normalizeClusterVersion = React.useCallback(
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply src/components/OpenshiftVersionsContextProvider.tsx:51]: Then backend validation fails.

**Review by jtomasek — APPROVED** [review]

**Pull request merged**